app/models/salt_event.rb

```diff
@@ -0,0 +1,48 @@
+class SaltEvent < ApplicationRecord
+  # Processing of a single event shouldn't take more than than 5 minutes.
+  # We consider events which were taken 5 minutes ago but not yet processed as
+  # if the worker died so we reassign them for processing.
+  PROCESS_TIMEOUT_MIN = 5
+  NO_EVENTS_TIMEOUT_SEC = 5
+
+  def self.process(worker_id:)
+    logger.info "Salt event processor #{worker_id} started at #{Time.current}"
+
+    loop do
+      sleep(NO_EVENTS_TIMEOUT_SEC) if !process_next_event(worker_id: worker_id)
+    end
+  end
+
+  def self.process_next_event(worker_id:)
+    taken_event = SaltEvent.where(
+      "processed_at IS NULL AND (worker_id IS NULL OR (worker_id = ?) OR taken_at < ?)",
```

Comment:
I'd try to use scopes here. Something like `SaltEvent.not_processed.taken_by_or_nobody(worker_id).taken_before(PROCESS_TIMEOUT_MIN.minutes.ago)`.